test/manual/mongodb_oidc.prose.test.ts

```diff
@@ -452,5 +452,153 @@ describe('MONGODB-OIDC', function () {
         });
       });
     });
+
+    // The driver MUST test reauthentication with MONGODB-OIDC for a read operation.
+    describe('6. Reauthentication', function () {
+      let refreshInvokations = 0;
+      let findStarted = 0;
+      let findSucceeded = 0;
+      let findFailed = 0;
+      let saslStarted = 0;
+      let saslSucceeded = 0;
+      let client;
+      let collection;
+      const cache = OIDC_WORKFLOWS.get('callback').cache;
+
+      // - Create request and refresh callbacks that return valid credentials that
+      //   will not expire soon.
+      const requestCallback = async () => {
+        const token = await readFile(`${process.env.OIDC_TOKEN_DIR}/test_user1`, {
+          encoding: 'utf8'
+        });
+        return { accessToken: token, expiresInSeconds: 300 };
+      };
+
+      const refreshCallback = async () => {
+        const token = await readFile(`${process.env.OIDC_TOKEN_DIR}/test_user1`, {
+          encoding: 'utf8'
+        });
+        refreshInvokations++;
+        return { accessToken: token, expiresInSeconds: 300 };
+      };
+
+      const commandStarted = event => {
+        if (event.commandName === 'find') {
+          findStarted++;
+        }
+        if (event.commandName === 'saslStart') {
+          saslStarted++;
+        }
+      };
+
+      const commandSucceeded = event => {
+        if (event.commandName === 'find') {
+          findSucceeded++;
+        }
+        if (event.commandName === 'saslStart') {
+          saslSucceeded++;
+        }
+      };
+
+      const commandFailed = event => {
+        if (event.commandName === 'find') {
+          findFailed++;
+        }
+      };
+
+      before(function () {
+        // - Clear the cache
+        cache.clear();
+        // - Create a client with the callbacks and an event listener capable of
+        //   listening for SASL commands
+        //
+        // - TODO(NODE-3494): Driver does not observe sensitive commands.
```

Comment:
Can you clarify this, comment? We don't observe sensitive commands, we do redact them, but we still emit command events for them (and that's how this test is passing).  given that the prose test only asserts that the events were emitted and nothing about the payload of the command, we could consider removing the comment entirely.

Reply:
I actually just need to delete the comment since this was implemented but the valid-pass tests in NODE-3494 aren't synced.